src/fe/fe.C

```diff
@@ -298,38 +298,58 @@ void FE<Dim,T>::reinit(const Elem * elem,
       {
         if (T != LAGRANGE)
           nonlagrange_dual_warning();
-        if (this->calculate_dual_coeff)
-        {
-          // In order for the matrices for the biorthogonality condition to be
-          // non-singular, the dual basis coefficients must be computed when the
-          // primal shape functions are evaluated with a quadrature rule. *But* a
-          // user may be reiniting with integration points from a mortar segment,
-          // which is valid, so a simple `if (!pts)` check is not
-          // appropriate. We're just gonna have to trust the user on this one. If
-          // they "screw up" we'll throw an exception from the LU decomposition,
-          // and they can choose to handle it or not
-
-          // instead of using the customized qrule from mortar segments,
-          // we need the default qrule for computing the dual coefficients
-          FEType default_fe_type(this->get_order(), T);
-          QGauss default_qrule(elem->dim(), default_fe_type.default_quadrature_order());
-          default_qrule.init(elem->type(), elem->p_level());
-          // In preparation of computing dual_coeff, we compute the default shape
-          // function values and save these in dual_phi (instead of declaring a
-          // new container). The TRUE dual_phi values are computed in
-          // compute_dual_shape_functions()
-          all_shapes(elem, default_fe_type.order, default_qrule.get_points(), this->dual_phi);
-          this->compute_dual_shape_coeffs(default_qrule);
-          // only compute the dual coefficient once
-          this->calculate_dual_coeff = false;
-        }
-        // the dual shape functions relies on the customized shape functions
+        // Check if we need to calculate the dual coefficients based on the default QRule
+        // We keep the default dual coeff calculation for the initial stage of the simulation,
+        // as well as interior elements that are fully in contact
+        // Currently, we re-compute dual_coeff for all msm elements by setting `calculate_default_dual_coeff' = false
+        // in `Assembly::reinitDual` and use the customized QRule for calculating the dual shape coefficients
```

Comment:
This comment is referring to MOOSE code "Assembly::reinitDual" and MOOSE jargon "msm". If you would like to keep it here, maybe just clarify that is the case?